bot.ts

```diff
@@ -80,38 +95,51 @@ const commands: SlashCommandPartial[] = [
 		],
 	},
 	{
-		name: "source-code",
+		name: CommandNames.sourceCode,
 		description: "Get a link to the bot's source code",
-	}
+	},
 ];
 
 class SpeedrunCom extends ApplicationCommandsModule {
-	@slash("game-community")
+	// @slash(CommandNames.gameCommunity)
 	async gameCommunity(i: ApplicationCommandInteraction) {
 		const userMention = i.option("mention");
-		if (!userMention) {
-			await i.defer();
-		}
+		const gameAbbreviation = i.option("game");
+		const userId = userMention && userMention instanceof User
+			? userMention.id
+			: undefined;
+		const timeoutId = setTimeout(async () => await i.defer(), TIMEOUT_LIMIT_MS);
 
 		const res = await fetch(
-			`https://www.speedrun.com/api/v1/games/${i.option("game")}`,
+			`https://www.speedrun.com/api/v1/games/${gameAbbreviation}`,
+			{ headers },
 		);
-		const game: v1SearchResult | undefined = (await res.json()).data;
-
-		if (!game) {
+		clearTimeout(timeoutId);
+		if (!res.ok && res.status !== 404) {
+			console.error(
+				"Unexpected speedrun.com response: ",
+				res.status,
+				await res.text(),
+			);
+			return i.reply({
+				content: `Speedrun.com returned an unexpected response: ${res.status}`,
```

Comment:
```ts
`${userMention}, Speedrun.com returned an error. Status: \`${res.status}\`, Message: \`${res.statusText}\``
```

I think the error message should be something along the lines of this. Since the statusText makes it more non-developer understandable.

We could also have it so it returns the error producted by SRC, but if we can't access that, then it could use the statusText